app/controllers/dashboards_controller.rb

```diff
@@ -7,7 +7,16 @@ def index
   end
 
   def build
-    @version = InstitutionBuilder.run(current_user)
+    results = InstitutionBuilder.run(current_user)
+
+    @version = results[:version]
+    @error_msg = results[:error_msg]
+
+    if @error_msg.present?
+      flash.alert = "Preview Data not built: #{@error_msg}"
+    else
+      flash.notice = "Preview Data (#{@version.number}) built successfully"
+    end
```

Comment:
So this is one reason I am not crazy about custom actions like "build" and prefer just the standard resource based actions -- to me this is really unclear which template will be rendered. 

First of all, this should probably be a POST or PUT (is it?) Any time you are modifying data you should have POST or PUT (not GET). 

Does build have its own template that gets rendered? I would think you get redirected back to dashboards / index but that doesn't appear to be implicit. Generally speaking when a non-idempotent action results in an error, the template that called gets rendered again with the errors. When it is successful, then you get redirected.

You might want to use flash.now with render_and_return , and flash.notice redirect_to as the else case,